### PR TITLE
ml.prevete.Daty: 1.0alpha release

### PR DESCRIPTION
**Pull request by tallero**
Can you help me tag the unnecessary finish-args? At the moment the app would just need display windows, read data into its XDG directories and connect to internet.

### CODE REVIEW

File: org.prevete.Daty.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+    "app-id": "org.prevete.Daty",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.30",
+    "sdk" : "org.gnome.Sdk",
+    "tags" : [
```
Comment (bilelmoussaoui) [inline org.prevete.Daty.json]: Not needed.

File: org.prevete.Daty.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            "name" : "libhandy",
+            "buildsystem" : "meson",
+            "config-opts": [
+                "-Dglade_catalog=disabled"
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://source.puri.sm/Librem5/libhandy.git"
```
Comment (bilelmoussaoui) [inline org.prevete.Daty.json]: You should be building a stable release.

**Comment by bilelmoussaoui** [comment]
testbot, build org.prevete.Daty

**Comment by FakeShemp** [comment]
Is prevete.org down or is it a temporary URL?

File: org.prevete.Daty.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        "python3-beautifulsoup4.json",
+        "python3-bleach.json",
+        "python3-pywikibot.json",
+        "python3-requests.json",
+        "python3-setproctitle.json",
+        {
+            "name" : "libhandy",
+            "buildsystem" : "meson",
+            "config-opts": [
+                "-Dglade_catalog=disabled"
```
Comment (bilelmoussaoui) [inline org.prevete.Daty.json]: You should disable examples, tests, vapi too here.

**Comment by tallero** [comment]
@FakeShemp temporary

File: org.prevete.Daty.json [hunk 4]
```diff
@@ -1,11 +1,5 @@
 {
-    "app-id": "org.prevete.Daty",
-    "runtime" : "org.gnome.Platform",
```
Comment (bilelmoussaoui) [inline org.prevete.Daty.json]: You still need the runtime, the runtime-version, the sdk too. it's just the tags that I meant;

File: org.prevete.Daty.json (hunk 3 above)
Comment (bilelmoussaoui) [reply org.prevete.Daty.json]: vapi is enabled by default and you don't need that for a python app.

File: org.prevete.Daty.json (hunk 4 above)
Comment (tallero) [reply org.prevete.Daty.json]: Sorry, I have not found documentation on tags, so I thought of them of generic tags.

Comment (bilelmoussaoui) [reply org.prevete.Daty.json]: Tags are not used here on Flathub at all unless you want to tag the software as proprietary.

**Comment by nedrichards** [comment]
bot, build ml.prevete.Daty
testbot, build ml.prevete.Daty

**Comment by nedrichards** [comment]
@tallero this build will fail - your app-id and the name of the json file need to be the same (it's a flathub thing).

**Comment by nedrichards** [comment]
bot, build ml.prevete.Daty
testbot, build ml.prevete.Daty

**Comment by tallero** [comment]
@barthalion It took time to do nice screenshots and an overview picture for the repo. Apparently I can't format the description nor links, am I right?

File: appstream.xml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        </screenshot>
+    </screenshots>
+    <releases>
+        <release version="1.0a" date="2019-01-25">
+            <description>
+                <p>Preview: Reader functions only.</p>
+            </description>
+        </release>
+    </releases>
+</component>
```
Comment (barthalion) [inline appstream.xml]: Also needs OARS rating: https://hughsie.github.io/oars/

Ideally it should be shipped with your code so other distributions can also use it but it's okay for now.

**Comment by barthalion** [comment]
testbot, build ml.prevete.Daty

**Comment by nedrichards** [comment]
testbot, build ml.prevete.Daty
bot, build ml.prevete.Daty

**Comment by nedrichards** [comment]
testbot, build ml.prevete.Daty
bot, build ml.prevete.Daty

**Comment by nedrichards** [comment]
@tallero you should ideally put the appdata into your project upstream and install it there as part of a new release - so it stays up to date and is part of your project.

In this case I just quickly fixed it so the file was declared correctly within the manifest and installed from there.

File: ml.prevete.Daty.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      </description>
+    </release>
+  </releases>
+  <content_rating type="oars-1.0"/>
+
+  <!--FIXME: You can use a project or developer name if there's no company-->
+  <developer_name>Pellegrino Prevete</developer_name>
+
+  <!--FIXME: where to submit translations-->
+  <url type="translate"><!-- http://www.homepage.com/how-to-submit-translations.html --></url>
```
Comment (nedrichards) [inline ml.prevete.Daty.appdata.xml]: you should remove this if it's not available.

File: ml.prevete.Daty.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+<component type="desktop">
+  <id>ml.prevete.Daty</id>
+  <launchable type="desktop-id">ml.prevete.Daty.desktop</launchable>
+  <name>Daty</name>
+  <summary>Free Wikidata editor.</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0</project_license>
+  <description>
+    <p>
+      <b>Daty is a Wikidata editor</b>
```
Comment (nedrichards) [inline ml.prevete.Daty.appdata.xml]: Something a bit better here would be good. How about this (lightly adapted from the README)
> Daty is a free Wikidata editor adhering to GNOME Human Interface Guidelines, intended to enable a better editing workflow and faster deployment of requested user features for Wikidata.
Use Daty to search, select, read, batch edit items, script actions, share, visualize proposed changes and bots.

**Comment by nedrichards** [comment]
Builds fine now: https://flathub.org/builds/#/builders/2/builds/7726

File: ml.prevete.Daty.appdata.xml (hunk 7 above)
Comment (tallero) [reply ml.prevete.Daty.appdata.xml]: Thank you!

**Comment by nedrichards** [comment]
testbot, build ml.prevete.Daty
bot, build ml.prevete.Daty

**Comment by barthalion** [comment]
It does not start for me:

```

**Comment by tallero** [comment]
@barthalion I noticed that.
Apparently I forgot to change the reverse-dns in the user setup when fixing the domain for the release. I have fixed that and moved the version tag to actual HEAD, so recompiling now should produce a good build.
I sincerely hope this is the last test build.

**Comment by nedrichards** [comment]
@tallero cool! when you push an update here i'll kick off a build

**Comment by tallero** [comment]
@nedrichards it is working, you can test!

**Comment by barthalion** [comment]
testbot, build ml.prevete.Daty
bot, build ml.prevete.Daty

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/ml.prevete.Daty

Thanks!

**Comment by tallero** [comment]
@barthalion Should I have the permission to push changes to that repository?

**Comment by barthalion** [comment]
Yes, although Github is a bit confusing about it. I invited you to collaborators; you should have received the invitation. It has to be accepted before you're granted write access.

**Comment by barthalion** [comment]
According to documentation, https://github.com/flathub/ml.prevete.Daty/invitations should let you accept it.

**Pull request closed without merging**